tests/src/test/java/org/eclipse/hono/tests/http/TelemetryHttpIT.java

```diff
@@ -125,4 +127,102 @@ public void testUploadMessageFailsForLargePayload(final VertxTestContext ctx) {
             // THEN the message gets rejected by the HTTP adapter with a 413
             .onComplete(ctx.completing());
     }
+
+    /**
+     * Verifies that the upload of a telemetry message fails with a 503 status code
+     * when there is no consumer.
+     *
+     * @param ctx The test context
+     */
+    @Test
+    public void testUploadMessageFailsForNoConsumer(final VertxTestContext ctx) {
+
+        // GIVEN a device
+        final Tenant tenant = new Tenant();
+
+        helper.registry
+                .addDeviceForTenant(tenantId, tenant, deviceId, PWD)
+                .compose(ok -> {
+
+                    // WHEN the device tries to upload a telemetry message while there is no consumer for it
+                    final MultiMap requestHeaders = MultiMap.caseInsensitiveMultiMap()
+                            .add(HttpHeaders.CONTENT_TYPE, "text/plain")
+                            .add(HttpHeaders.AUTHORIZATION, authorization);
+
+                    return httpClient.create(
+                            getEndpointUri(),
+                            Buffer.buffer("hello"),
+                            requestHeaders,
+                            ResponsePredicate.status(HttpURLConnection.HTTP_UNAVAILABLE));
+
+                })
+                // THEN the message gets rejected by the HTTP adapter with a 503
+                .onComplete(ctx.succeeding(response -> {
+                    ctx.verify(() -> {
+                        assertThat(response.bodyAsString()).isEqualTo("temporarily unavailable");
```

Comment:
I mainly added this here so that with the fix for #2166, the more specific error message can be checked upon here.

I have removed the check now and will add it again with the fix for #2166 (using a constant then).